app/models/user.rb

```diff
@@ -577,11 +577,30 @@ def banned_at_date?(date)
     delegate_roles.select { |role| role.metadata.status == RolesMetadataDelegateRegions.statuses[:senior_delegate] }
   end
 
-  private def groups_with_read_access
+  private def can_view_current_banned_competitors?
+    wdc_team? || ethics_committee? || board_member? || staff_delegate? || weat_team? || results_team? || admin?
+  end
+
+  private def can_view_past_banned_competitors?
+    wdc_team? || ethics_committee? || board_member? || weat_team? || results_team? || admin?
+  end
+
+  private def groups_with_read_access_for_current
     return "*" if can_edit_any_groups?
     groups = groups_with_edit_access
 
-    if can_view_banned_competitors?
+    if can_view_current_banned_competitors?
+      groups += UserGroup.banned_competitors.ids
+    end
+
+    groups
+  end
```

Comment:
Design decision / code philosophy: Should the list of groups with "current" access also include the list of groups with "past" access?

Background: If you have edit access, the groups are also considered for (both cases of) read access.
But if you have full (i.e. past) read access, the group is not considered for restricted (i.e. current) read access.
In my mind, `edit_access > past_read_access > current_read_access`, so every permission group should also consider "higher" permissions. Is this feasible? Or is there some edge case that I'm not considering properly here?

Reply:
Agreed, this should cover all cases probably. Cannot think of any edge case where this fails. If somebody can see a past role, they can definitely see the current role.

Reply:
If you agree, then `groups_with_read_access_for_current` should probably reference `groups_with_read_access_for_past`. But it currently doesn't...

Reply:
Oh yes. Added now. I initially added only to `can_view_current_banned_competitors?`.